chapters/2.md

```diff
@@ -1,211 +1,170 @@
-* Show me, for a specific component, what events it sends out
-* Show me, for a specific component, what events it receives.
-
-This is completely 100% possible and viable. I just haven't had time to do it, yet, and decided to type this up first.
-
-## Conclusion
-
-If you want the massive performance, forward compatibility, and complete portability upsides of Polymer with none of the downsides - i.e., if you want to be able to ship things in Polymer/Web Components that are fast and that people use - this is what needs to happen.
-
-**There must be effective teaching material.** And that's what this Polymer Knowledge Center is: teaching-focused documentation, aimed at taking a developer from zero to their first production bug fix as quickly as possible. There must be a known, useful escalation chain so developers don't spend time *sitting at stuck* - even if that just means knowing who to ask and being able to do it.
-
-**Adopt Redux for state management.** Reduce the number of things a programmer has to hold in their head to make a change from (n^3) to (2 + easy). Let your QA people export bugs directly to your developers.
-
-**Build better tooling.** The only reason there isn't a Polymer browser plugin already is because all the Google engineers are great at using the default Chrome inspector. But it would help streamline the development and understanding of Polymer development by just taking what's already there, already accessible, already happening, and just making it more clear and easy to get to.
-
-Polymer/Web components aren't just one among many choices, they are the *only* choice. Because if most of the people who'll be trying to consume online content *can't do it through the browser*, then something else will fill that need. And that means, it doesn't matter if you know React. It doesn't matter if you know Angular. People won't be using a browser anyway.
-
-But more than that, I think Polymer/Web component based development can be an order of magnitude more effective than their JavaScript-framework based counterparts, and the distance between here and there is only three things.
+What is a template? Where have you seen the term in other areas of life? There are
+drawing templates, cookie templates and many more. How are these templates used?
+When I take a drawing template and I trace the edges provided by the template I
+always get the same resulting drawing afterward. For example, a template outlining
+a house will always result in a house. You won't draw a house four times followed
+by a giraffe (unless something odd is going on :-P). Templates in the polymer realm
+are conceptually the same. The `<template>` element in the polymer realm specifies
+an outline for the module content so that every time the custom element is seen
+in an HTML file the same content will be stamped in place of its name.
+
+TODO The template actually refers to template in transpiler terms. Include here?
```

Comment:
Key concepts about templates in my opinion:
* They're inert. They're a blueprint, that then gets inflated with information and placed into a browser.
* One of the main responsibilities of dom-module is to produce reliable, consistent, predictable access to the `<template>`
* Definitely mention this isn't a polymer-specific thing, but a platform thing. No more details needed, but in terms of developing a mental map, these are key pieces of info imo

Reply:
I actually discuss the template concept in terms of a drawing template. So, the first point should already be present though let me know if it's unclear. The second point is too verbose in my opinion. I express that in a low cognitive load fashion already. But, again, let me know if that doesn't across. And the third point is covered by the closing paragraph.

Reply:
w/r/t third point up there, just got to end of chapter. nice!!! nix that

Reply:
ha you beat me to it

Reply:
This may be too much to include this early in the chapters but I still don't understand why we have <template></template> in the framing of the style/html in every Polymer file but then we also have these random <template></template> tags with nothing else in them thrown into certain elements and if you ignore them, bad things will happen. And then <template> is also part of the dom-if and dom-repeat constructions and so what is the commonality between these three instances?